### PR TITLE
Harry/3349 no generic extensions

### PR DESCRIPTION
**Pull request by hwh33**
Avoids the use of generic extensions in `FingerprintClientHello` as discussed in [this thread](https://github.com/getlantern/utls/pull/10#discussion_r480377064).

### CODE REVIEW

File: u_tls_extensions.go [hunk 1]
```diff
@@ -718,7 +718,7 @@ func (e *FakeCertCompressionAlgsExtension) Read(b []byte) (int, error) {
 	if len(b) < e.Len() {
 		return 0, io.ErrShortBuffer
 	}
-	// https://tools.ietf.org/html/draft-balfanz-tls-channelid-00
+	// https://tools.ietf.org/html/draft-ietf-tls-certificate-compression-10#section-3
```
Comment (hwh33) [inline u_tls_extensions.go:721]: Drivey-by - I think this was just an errant copy/paste

File: u_tls_extensions.go [hunk 2]
```diff
@@ -727,8 +727,11 @@ func (e *FakeCertCompressionAlgsExtension) Read(b []byte) (int, error) {
 		return 0, errors.New("too many certificate compression methods")
 	}
 
+	// Extension data length.
 	b[2] = byte((extLen + 1) >> 8)
 	b[3] = byte((extLen + 1) & 0xff)
+
+	// Methods length.
```
Comment (hwh33) [inline u_tls_extensions.go:734]: The structure of the code here confused me, so I added a couple comments and a newline

**Comment by hwh33** [comment]
They are being tested, and yes it is because of the hellos in the line you linked.  `HelloFirefox_65` tests the record size limit extension and `HelloChrome_70` (amongst others) tests the cert compression extension.  I know because I had to fix my bugs to pass the tests =). Those tests are nicely thorough by the way!  Very helpful

**Comment by hwh33** [comment]
Sorry, `HelloFirefox_65` isn't in the list, but it's equivalent to `HelloFirefox_63` which is tested

**Comment by max-b** [comment]
> They are being tested, and yes it is because of the hellos in the line you linked. `HelloFirefox_65` tests the record size limit extension and `HelloChrome_70` (amongst others) tests the cert compression extension. I know because I had to fix my bugs to pass the tests =). Those tests are nicely thorough by the way! Very helpful

Great thanks!!! :+1:

**Comment by hwh33** [comment]
I'm going to go ahead and merge.  Thanks for the review @max-b!

**Pull request merged**